pkg/controller/direct/bigquerydataset/dataset_controller.go

```diff
@@ -0,0 +1,323 @@
+	}
+	if !reflect.DeepEqual(desired.FriendlyName, resource.FriendlyName) {
+		resource.FriendlyName = desired.FriendlyName
+		updateMask.Paths = append(updateMask.Paths, "friendly_name")
+	}
+	if !reflect.DeepEqual(desired.DefaultPartitionExpirationMs, resource.DefaultPartitionExpirationMs) {
+		resource.DefaultPartitionExpirationMs = desired.DefaultPartitionExpirationMs
+		updateMask.Paths = append(updateMask.Paths, "default_partition_expirationMs")
+	}
+	if !reflect.DeepEqual(desired.DefaultTableExpirationMs, resource.DefaultTableExpirationMs) {
+		resource.DefaultTableExpirationMs = desired.DefaultTableExpirationMs
+		updateMask.Paths = append(updateMask.Paths, "default_table_expirationMs")
+	}
+	if !reflect.DeepEqual(desired.DefaultCollation, resource.DefaultCollation) {
+		resource.DefaultCollation = desired.DefaultCollation
+		updateMask.Paths = append(updateMask.Paths, "default_collation")
+	}
+	if desired.DefaultEncryptionConfiguration != nil && resource.DefaultEncryptionConfiguration != nil && !reflect.DeepEqual(desired.DefaultEncryptionConfiguration, resource.DefaultEncryptionConfiguration) {
+		// Resolve KMS key reference
+		if a.desired.Spec.DefaultEncryptionConfiguration != nil {
+			kmsRef, err := refs.ResolveKMSCryptoKeyRef(ctx, a.reader, a.desired, a.desired.Spec.DefaultEncryptionConfiguration.KmsKeyRef)
+			if err != nil {
+				return err
+			}
+			desired.DefaultEncryptionConfiguration.KmsKeyName = kmsRef.External
+		}
+		resource.DefaultEncryptionConfiguration.KmsKeyName = desired.DefaultEncryptionConfiguration.KmsKeyName
+		updateMask.Paths = append(updateMask.Paths, "default_encryption_configuration")
+	}
+	if !reflect.DeepEqual(desired.IsCaseInsensitive, resource.IsCaseInsensitive) {
+		resource.IsCaseInsensitive = desired.IsCaseInsensitive
+		updateMask.Paths = append(updateMask.Paths, "is_case_sensitive")
+	}
+	if !reflect.DeepEqual(desired.MaxTimeTravelHours, resource.MaxTimeTravelHours) {
+		resource.MaxTimeTravelHours = desired.MaxTimeTravelHours
+		updateMask.Paths = append(updateMask.Paths, "max_time_interval_hours")
+	}
+	if desired.Access != nil && resource.Access != nil && len(desired.Access) > 0 && !reflect.DeepEqual(desired.Access, resource.Access) {
+		for _, access := range desired.Access {
+			resource.Access = append(resource.Access, access)
+		}
+		updateMask.Paths = append(updateMask.Paths, "access")
+	}
+	if !reflect.DeepEqual(desired.StorageBillingModel, resource.StorageBillingModel) {
+		resource.StorageBillingModel = desired.StorageBillingModel
+		updateMask.Paths = append(updateMask.Paths, "storage_billing_model")
+	}
+
+	if len(updateMask.Paths) == 0 {
+		return nil
+	}
+	parent, datasetId, err := krm.ParseBigQueryDatasetExternal(a.id.External)
+	if err != nil {
+		return fmt.Errorf("failed to parse bigquery dataset full name, %w", err)
+	}
+
+	if desired.Access == nil || len(desired.Access) == 0 {
+		datasetGetCall := a.gcpService.Datasets.Get(parent.ProjectID, datasetId)
```

Comment:
Why do we need to call `Dataset.Get` again to fetch the Access list? Isn't it already in `a.actual / resource`?